### ISSUE TITLE
REVEAL: Adding the title Templates to the content type YAML

### ISSUE THREAD

**Issue by laceysanderson** [tripal/tripal#1563]
# Core Tripal 4 Development Task

## Pull #1583

**This issue may be as simple as uncommenting some lines in the content types YAML, bringing back a dockblock comment in the Services/TripalContentTypes::createContentType() and adding testing to [the existing test for that service](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/tests/src/Functional/Services/TripalContentTypesTest.php). Of course manual testing will also be needed.**

## Current Issue Description

Originally, title defaults were created by the Tripal Content Type entity in [TripalEntityType->getDefaultTitleFormat()](https://github.com/tripal/tripal/blob/4.x/tripal/src/Entity/TripalEntityType.php#L457-L514). Then any module could implement hook_tripal_default_title_format() to add additional suggestions.

While this works, it is more difficult to change defaults then it needs to be with updates introduced in Drupal 8.

Instead PR #1413, added the title default per content type to the content type YAML file (they are still there but commented out currently) and then used that information to create the defaults.

**The following code links are in the original branch and show how it may have been implemented:**
The [TripalEntityType->getDefaultTitleFormat()](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/src/Entity/TripalEntityType.php#L468-L529) remains unchanged, and when getting the title format, it still checks to see if the title format has been set and only if it hasn't is getDefaultTitleFormat() called.

The difference comes in creating the content types in the first place:
- Since the title format is now included in the YAML, you will see in the service creating content types that it is [passed in through the details](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/src/Services/TripalContentTypes.php#L167).
- This means it gets passed into [TripalEntityType::create() here](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/src/Services/TripalContentTypes.php#L196).
- And then passed into the [parent::create() here](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/src/Entity/TripalEntityType.php#L198) which essentially sets TripalEntityType::title_format which means that if it is set in the YAML then [TripalEntityType->getDefaultTitleFormat()](https://github.com/tripal/tripal/blob/tv4g1-1393-ContentTypesYML/tripal/src/Entity/TripalEntityType.php#L468-L529) will not get called.

**Bringing this functionality back may be as simple as commenting out the title formats in the Yaml and adding back in the documentation line linked above!**

## Background

The background for this issue is a cautionary tale: Once upon a time a PR was made, #1413, with so many great changes for Tripal 4. This PR was focused on moving content type + field creation into YAML files for the good of all. This was a large task in and of itself but as such tasks have a habit of doing... it picked up many additional bug fixes and features as it grew. The developer was testing locally both manually and running the automated tests they developed just as they should but forgot to push out to github as often as recommended 😨 This caused the developer to miss a test failure in unrelated code within their beautiful PR 😱 As the PR had become an inter-related collection of fixes by the point it came up for review and the unrelated failing test was found... it became a nightmare to debug and though the developer and their friends tried, they could not find the bug 😭 🗿  This meant that the PR had to be redone with careful testing and often github pushing to ensure the bug did not resurface... This processed birthed PR #1475 and this collection of issues.

tl;dr;
- PR #1413 included a large number of fixes but needed to be closed due to a test failure in unrelated code which could not be fixed
- PR #1475 brought back the core functionality the PR originally meant to cover (Content Types + Fields created via YAML) but could not bring back all of the fixes from the original PR.
- This issue has been created to mark the place of one of these fixes to be brought back.

## Process:
- Look at PR #1413 and try to find the code that made this particular fix. I will try to point at some of it in this issue to aid this process.
- Create a new branch from 4.x using the branch name above
- Copy small bits of the changes from PR #1413 into this branch and commit. Only move pieces related to the fix!
  - When committing you will want to give both yourself and Stephen credit for the code! See following for how to do that (Co-authored commit).
- Push to the branch after each addition to make sure there are no test failures 😬 

### Co-Authored Commits

See [these github docs for more detail](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

For each commit including any of Stephens original code use the following. Note that you do not close the quotation after your typical commit message but rather enter two empty lines and add the co-author information.

```bash
git commit -m "Your typical commit message


Co-authored-by: Stephen Ficklin <spficklin@gmail.com>
Co-authored-by: YOUR-NAME <YOUR-NAME@EXAMPLE.COM>"
```

Example: https://github.com/tripal/tripal/pull/1475/commits/af6bc9b18b484fa7d24eda73bf2b73b6256d57cd

If needed you can alter your commit message to add this after the fact. This is easiest if done before pushing to github.

**Comment by dsenalik** [comment]
* Oops, disregard, I built my docker incorrectly

**Comment by dsenalik** [comment]
Fixed with merged PR #1583
